plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/AlternateDeploymentMapper.java

```diff
@@ -0,0 +1,228 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.models;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import org.jboss.tools.common.databinding.ObservablePojo;
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistryAdapter;
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistrySingleton;
+import org.jboss.tools.openshift.common.core.connection.IConnection;
+import org.jboss.tools.openshift.common.core.connection.IConnectionsRegistryListener;
+import org.jboss.tools.openshift.core.OpenShiftAPIAnnotations;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.core.connection.ConnectionProperties;
+import org.jboss.tools.openshift.internal.core.Trace;
+import org.jboss.tools.openshift.internal.core.WatchManager;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.IService;
+
+public class AlternateDeploymentMapper extends ObservablePojo implements OpenShiftAPIAnnotations, IDeploymentResourceMapper{
```

Comment:
The class only exists because I knew the feature you are working on is using the other and I didnt want to cause you problems.  Ideally, the content from this class will completely replace the class content of the other and we utilize the previous class name.